### ISSUE TITLE
[Bug]: CalDavBackend->getChangesForCalendar() reports move operation events (added, deleted) on wrong calendars

### ISSUE THREAD

**Issue by ksainc** [nextcloud/server#40051]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

CalDavBackend->getChangesForCalendar() reports move operation events (added, deleted) on wrong calendars

Moved calendar object from calendar 26 (Combined) to 24 (Personal). Then ran CalDavBackend->getChangesForCalendar() on both.

Results:

Calendar 24 (Personal) shows delete event but should show added
```
$this->LocalStore->getChangesForCalendar(24, 132, null, null)
array(4)
    syncToken: 134  
    added: array(0)
    modified: array(0)
    deleted: array(1)
        0: "9ED4BC5B-EE08-4E02-8E49-B341CF5C7E64.ics"
```

Calendar 26 (Combined) shows added event but should show deleted
```
$this->LocalStore->getChangesForCalendar(26, 32, null, null)
array(4) 
    syncToken: 33  
    added: array(1)  
        0: "9ED4BC5B-EE08-4E02-8E49-B341CF5C7E64.ics"  
    modified: array(0)  
    deleted: array(0)  
```

### Steps to reproduce

1. Move event between two calendars
2. Execute CalDavBackend->getChangesForCalendar() on each calendar

### Expected behavior

Report move operation events (added, deleted) on proper calendar

### Installation method

None

### Nextcloud Server version

26

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Nginx

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by kesselb** [comment]
cc @ChristophWurst @miaulalala

**Comment by miaulalala** [comment]
Which version of 26 are you using? This might have been fixed in 26.0.2 already by https://github.com/nextcloud/server/pull/37812

**Comment by ksainc** [comment]
Good Morning @miaulalala,

Confirmed. Its been fixed. 

Thank you.
